runtime/src/lib.rs

```diff
@@ -34,6 +34,7 @@ pub mod stake_weighted_timestamp;
 pub mod stakes;
 pub mod static_ids;
 pub mod status_cache;
+pub mod svm_message_ext;
```

Comment:
Can we keep these extensions in the svm-transaction crate? Even though these methods are used by the runtime, they're still useful in other contexts. Off chain clients may want to check if a transaction uses a durable nonce for example.

Reply:
It'd add a dependency on bincode in `svm-transaction` trait, which is undesirable.

When I talked with @buffalojoec about this we discussed creating a separate crate for this derived behavior.
Not sure if we'd like to do that now, or wait until there's a larger set of derived behaviors.
Didn't feel right to create a new crate for 2 functions.

Reply:
another option is to rework the implementation to avoid bincode - would be relatively straight-forward since all we use it for is to read a 4-byte discriminant.

Reply:
@jstarry thoughts on which approach you think is best here?

Reply:
Just adding one more point to supplement my argument for the runtime/new crate. Lmk if it's off-base.

Even though nonce management is split between runtime and SVM right now, the concept of a durable nonce is actually more pertinent to the runtime, when checking transaction age. There's an argument to be made for actually making the act of advancing the nonce during transaction execution a plugin that extends SVM, since not everyone will use nonces in other execution environments.

Anyway that's my perspective on it. I think we should keep the SVM crate composable and as light as possible, opening up custom behavior through plugins implemented elsewhere rather than bloat.

Reply:
I agree with @buffalojoec that this is a runtime concept and doesn't make sense for it to live in the svm transaction crate. I think it's going to all make much more sense when we have a "runtime transaction" abstraction